jmx_prometheus_httpserver/src/main/java/io/prometheus/jmx/WebServer.java

```diff
@@ -28,14 +26,7 @@ public static void main(String[] args) throws Exception {
        socket = new InetSocketAddress(port);
      }
 
-     JmxCollector jc = new JmxCollector(new File(args[1])).register();
-
-     Server server = new Server(socket);
-     ServletContextHandler context = new ServletContextHandler();
-     context.setContextPath("/");
-     server.setHandler(context);
-     context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
-     server.start();
-     server.join();
+     new JmxCollector(new File(args[1])).register();
+     new HTTPServer(socket, CollectorRegistry.defaultRegistry);
```

Comment:
Not really, HTTPServer creates threads that will keep running, even if the main thread exits, c.f. [termination of main thread exit](https://stackoverflow.com/questions/5642802/termination-of-program-on-main-thread-exit).

You can run the [PR locally as well](https://help.github.com/articles/checking-out-pull-requests-locally/).